crates/client/src/client/schema/tx/transparent_tx.rs

```diff
@@ -80,13 +82,21 @@ pub struct Transaction {
     pub id: TransactionId,
     pub tx_pointer: Option<TxPointer>,
     pub input_asset_ids: Option<Vec<AssetId>>,
+    /// The list of all contracts from the inputs of the transaction.
+    ///
+    /// It is a helper function and doesn't have a corresponding field in the transaction.
     pub input_contracts: Option<Vec<ContractIdFragment>>,
+    /// The field of the `Mint` transaction.
```

Comment:
the `input_contract` is a field of the `Mint` transaction?

Reply:
Yes

Reply:
Sorry. I didn't phrase this well enough. I don't understand what it means for `input_contracts` to be a field on `Mint`. Same with `output_contract`. It's a field on `Transaction`.

Reply:
<img width="795" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/1d39b135-7f29-4f7c-bfee-ab100cb910fa">

Reply:
Okay. They both have that field. They both have the `tx_pointer` field as well. I guess my point is when I'm reading through the `Transaction` code and I see the comment `The field of the `Mint` transaction.`, I'm not sure why this is important. If it is _only_ used by mints, then perhaps name it `mint_input_contract`, etc. If it serves multiple purposes, but it's important for the reader to understand that it is _also_ used on the `Mint` transaction, then maybe we could explain in the comment why it's important we take note of that.

The way it is written confused me, so I assume it will confuse someone else as well.

Reply:
I agree that a name like `mint_input_contract` would be more clear. You could also add a comment that describes the relationship between transparent txs and mint and how the input/output contracts represent that relationship.

Reply:
Maybe adding docs for the overall struct explaining why it's constructed of a collage of different transaction type parts would be good. Other than that, the docs you've added help a ton!